examples/vae/train_vae.py

```diff
@@ -3,154 +3,120 @@
+                        help='Use tiny datasets for quick tests')
+    args = parser.parse_args()
+
+    print('GPU: {}'.format(args.gpu))
+    print('# dim z: {}'.format(args.dimz))
+    print('# Minibatch-size: {}'.format(args.batchsize))
+    print('# epoch: {}'.format(args.epoch))
+    print('')
+
+    # Prepare VAE model, defined in net.py
+    model = net.VAE(784, args.dimz, 500)
+    if args.gpu >= 0:
+        chainer.cuda.get_device_from_id(args.gpu).use()
+        model.to_gpu()
+
+    # Setup an optimizer
+    optimizer = chainer.optimizers.Adam()
+    optimizer.setup(model)
+
+    # Initialize
+    if args.initmodel:
+        chainer.serializers.load_npz(args.initmodel, model)
+
+    # Load the MNIST dataset
+    train, test = chainer.datasets.get_mnist(withlabel=False)
+    if args.test:
+        train, _ = chainer.datasets.split_dataset(train, 100)
+        test, _ = chainer.datasets.split_dataset(test, 100)
+
+    train_iter = chainer.iterators.SerialIterator(train, args.batchsize)
+    test_iter = chainer.iterators.SerialIterator(test, args.batchsize,
+                                                 repeat=False, shuffle=False)
+
+    # Set up an updater. StandardUpdater can explicitly specify a loss function
+    # used in the training with 'loss_func' option
+    updater = training.StandardUpdater(train_iter, optimizer, device=args.gpu,
+                                       loss_func=model.get_loss_func())
+
+    trainer = training.Trainer(updater, (args.epoch, 'epoch'), out=args.out)
+    trainer.extend(extensions.Evaluator(test_iter, model, device=args.gpu,
+                                        eval_func=model.get_loss_func(k=10)))
+    trainer.extend(extensions.dump_graph('main/loss'))
+    trainer.extend(extensions.snapshot(), trigger=(args.epoch, 'epoch'))
+    trainer.extend(extensions.LogReport())
+    trainer.extend(extensions.PrintReport(
+        ['epoch', 'main/loss', 'validation/main/loss',
+         'main/rec_loss', 'validation/main/rec_loss', 'elapsed_time']))
+    trainer.extend(extensions.ProgressBar())
+
+    if args.resume:
+        chainer.serializers.load_npz(args.resume, trainer)
+
+    # Run the training
+    trainer.run()
+
+    # Visualize the results
+    def save_images(x, filename):
+        import matplotlib.pyplot as plt
+        fig, ax = plt.subplots(3, 3, figsize=(9, 9), dpi=100)
+        for ai, xi in zip(ax.flatten(), x):
+            ai.imshow(xi.reshape(28, 28))
+        fig.savefig(filename)
+
+    model.to_cpu()
+    train_ind = [1, 3, 5, 10, 2, 0, 13, 15, 17]
+    x = chainer.Variable(np.asarray(train[train_ind]))
+    with chainer.no_backprop_mode():
+        x1 = model(x)
+    save_images(x.data, os.path.join(args.out, 'train'))
+    save_images(x1.data, os.path.join(args.out, 'train_reconstructed'))
+
+    test_ind = [3, 2, 1, 18, 4, 8, 11, 17, 61]
+    x = chainer.Variable(np.asarray(test[test_ind]))
     with chainer.no_backprop_mode():
```

Comment:
The network used here doesn't have any layers which switch their behavior by `chainer.config.train`, but users might modify the network and use this training script as it is. So, how about adding `chainer.using_config('train', False)` context along with this? You can specify both by `with chainer.using_config('train', False), chainer.no_backprop_mode()`.